src/gas_price_increase.rs

```diff
@@ -39,11 +39,12 @@ pub fn enforce_minimum_increase_and_cap(
     let mut last_used_gas_price = None;
     stream.filter_map(move |gas_price| {
         assert!(gas_price.is_finite() && gas_price >= 0.0);
-        let gas_price = if let Some(previous) = last_used_gas_price {
-            new_gas_price_estimate(previous, gas_price, gas_price_cap)
-        } else {
-            Some(gas_price)
```

Comment:
I think the change is a little more obvious by changing this line to

```rust
Some(std::cmp::min(gas_price, gas_price_cap))
```

than to use -1.0 as the place holder. Because new_gas_price_estimate might assume that a price is never negative.

I'm fine with your change too, just a suggestion.

Reply:
I also prefer :point_up: